lib/parsers/m3u_parser.rb

```diff
@@ -0,0 +1,22 @@
+class FormatParser::M3UParser
+  include FormatParser::IOUtils
+
+  HEADER = '#EXTM3U'
+
+  def likely_match?(filename)
+    filename =~ /\.m3u8?$/i
+  end
+
+  def call(io)
+    io = FormatParser::IOConstraint.new(io)
+
+    header = safe_read(io, 7)
+    return unless HEADER.eql?(header)
+
+    FormatParser::Text.new(
+      format: :m3u,
+      size: io.size
```

Comment:
Just want to be sure: will this read the whole file in order to get the size? Because if it does, it might be wiser not to include it (or implement it in a different way).

Reply:
`size` is available once a single read has been performed and will not incur any extra reads